core/iwasm/common/wasm_runtime_common.c

```diff
@@ -1309,7 +1309,7 @@ wasm_runtime_finalize_call_function(WASMExecEnv *exec_env,
 
     bh_assert((argv && ret_argv) || (argc == 0));
 
-    if (argv == ret_argv || argc == 0) {
```

Comment:
Not sure why remove it, will it cause issue?

Reply:
Yes, it will. If there is an `externref` in parameters, the code will malloc `new_argv` to transfer, and `new_argv` should be freed even if there is no result.

Reply:
OK